org.eclipse.dawnsci.analysis.api/src/org/eclipse/dawnsci/analysis/api/persistence/IPersistentFile.java

```diff
@@ -254,12 +253,13 @@ default public void setAxes(IDataset... axes) throws Exception {
 	 * Method that reads a List of axes from entry/data.<br>
 	 * 
 	 * @param mon
+	 * @param dataName can be null for first data
 	 * @param axisNames
 	 * @return List<ILazyDataset>
 	 * @throws Exception
 	 *              is thrown if no correct entry is found in the file
 	 */
-	default public List<ILazyDataset> getAxes(IMonitor mon, String... axisNames) throws Exception {
+	default public List<ILazyDataset> getAxes(IMonitor mon, String dataName, String... axisNames) throws Exception {
 		if (axisNames == null || axisNames.length == 0) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reorder the modifiers to comply with the Java Language Specification. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)